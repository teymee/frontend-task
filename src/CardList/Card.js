import React from "react";

function Card({name, description}) {
	return (
		<div className="card  my-10">
			<div className="container py-5">
				<h1 className="md:text-lg font-semibold lg:text-2xl">{name}</h1>
				<p className="my-5 md:text-sm lg:text-lg">
					{description}
				</p>
			</div>
			{/* <p>{date}</p> */}

                  <span className="pl-9 py-3">Use Template</span>
		</div>
	);
}

export default Card;
