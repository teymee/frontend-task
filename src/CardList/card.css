.cardList {
	margin: auto auto;
	

	/* margin-left: 200px; */
}


.card {
	background: #ffffff;
	/* background-color: #1e1e1e; */
	/* color: #b9b9b9; */
	box-shadow: 0px 4px 30px 0px #00000014;
	/* width: 80%; */
	border-radius: 8px;
	/* height: 200px; */
}

.header {
	/* width: 90%; */
	margin: auto auto;
}

.card .container {
	width: 80%;
	height: 75%;
	margin: auto auto;
}

.card span {
	background: #f9f9f9;
	/* background-color: #2d2d2d; */
	/* color: #c9f7f2; */
	width: 100%;
	display: block;
	color: #08BD37;
	font-size: 13px;
	height: 25%;
	border-radius: 8px;
}

.card h1 {
	/* font-size: 24px; */
	font-weight: 500;
	line-height: 32px;
	letter-spacing: 0em;
	text-align: left;
}

.card p {
	font-family: Inter;
	font-size: 13px;
	font-weight: 400;
	line-height: 20px;
	letter-spacing: 0em;
	text-align: left;
}

/* PAGINATION */

.pagination {
	margin: auto auto;
	/* width: 70% !important; */
	padding-bottom: 200px;
	display: flex;
	justify-content: space-around;
	/* background-color: blue; */
}

.indicator {
	font-weight: bold;
}

.active-page {
	background-color: black;
	color: white;
	padding: 0.5rem 1rem;
	border-radius: 5px;
}

@media only screen and (max-width: 600px) {
	.pagination {
		width: 100% !important;
	}

	.active-page {
		padding: 0.2rem 0.5rem;
		border-radius: 5px;
	}
}
