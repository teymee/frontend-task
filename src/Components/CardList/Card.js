import React from "react";

function Card({ details }) {
	return (
		<div className="card  my-10">
			<div className="container py-5">
				<h1 className="md:text-lg font-semibold lg:text-2xl">{details.name}</h1>
				<p className="my-5 md:text-sm lg:text-lg">{details.description}</p>
			</div>

			<span className="pl-9 py-3">Use Template</span>
		</div>
	);
}

export default Card;
