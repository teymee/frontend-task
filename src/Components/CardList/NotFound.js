import React from "react";

import Empty from "./../../assets/empty.svg";

function NotFound() {
	return (
		<>
			<div className="flex justify-center align-center lg:h-80 md:h-40  mt-40">
				<img src={Empty} alt="empty" />
			</div>
			<h4 className="text-center pt-10 pb-20 text-lg font-bold">
				No Record Found
			</h4>
		</>
	);
}

export default NotFound;
