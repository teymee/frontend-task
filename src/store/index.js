import {configureStore} from '@reduxjs/toolkit' 
import TemplateSlice from './template-slice';

const store = configureStore({
      reducer: {template: TemplateSlice.reducer},
     
})

export default store;