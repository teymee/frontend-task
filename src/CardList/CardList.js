import React, { useState } from "react";
import { useSelector } from "react-redux";
import ReactPaginate from "react-paginate";

import Card from "./Card";
import "./card.css";

function CardList() {
	const { templates, loading, categorizedTemp, tempCategory } = useSelector(
		(state) => state.template
	);

	const [curPage, setCurPage] = useState(0);

	let pageCount = 15;

	const pageChange = ({ selected }) => {
		return setCurPage(selected);
	};

	let card, pagNum, finalData;
	if (categorizedTemp !== null) {
		finalData = categorizedTemp;

		let pageVisited = curPage * pageCount;

		let tempList = finalData.slice(pageVisited, pageVisited + pageCount);

		pagNum = Math.ceil(finalData.length / pageCount);
		card = tempList.map((item) => {
			return (
				<Card
					name={item.name}
					description={item.description}
					link={item.link}
					date={item.created}
				/>
			);
		});
	}

	return (
		<>
			<div className=" mt-20 flex justify-between header " >
				<h5 data-testid="test-1 " className="text-md ">{tempCategory} Templates</h5>
				<h5 className="text-slate-400 text-sm">
					{finalData !== undefined && finalData.length} Templates
				</h5>
			</div>
			<div className=" grid md:grid-cols-3 lg:gap-x-24 md:gap-5 sm:grid-cols-2   cardList">
				{finalData !== null && card}
			</div>
			<ReactPaginate
				previousLabel={curPage === 0 ? "Previous" : "< Previous"}
				nextLabel={curPage === pagNum - 1 ? "Next" : "Next"}
				breakLabel="..."
				onPageChange={pageChange}
				pageCount={pagNum}
				pageRangeDisplayed={2}
				marginPagesDisplayed={2}
				containerClassName="pagination md:w-1/4  text-sm py-10 lg:pt-20"
				previousClassName="indicator "
				nextClassName="indicator"
				activeLinkClassName="active-page"
			/>
		</>
	);
}

export default CardList;
