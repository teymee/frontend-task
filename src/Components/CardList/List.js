import React, { useState } from "react";
import Card from "./Card";
import "./card.css";
import Pagination from "./Pagination";
import { useSelector } from "react-redux";

function List() {
	const [curPage, setCurPage] = useState(0);

	// PAGE CHANGE
	const pageChange = ({ selected }) => {
		return setCurPage(selected);
	};

	// REDUXTOOLKIT STATES
	const { categorizedTemp, tempCategory } = useSelector(
		(state) => state.template
	);

	// INITIALIZATIONS
	let card, pagNum, pageCount, finalData;
	finalData = categorizedTemp;
	pageCount = 15;

	// PAGINATION CODE-BLOCK
	if (finalData !== null) {
		let pageVisited = curPage * pageCount;
		let tempList = finalData.slice(pageVisited, pageVisited + pageCount);
		pagNum = Math.ceil(finalData.length / pageCount);
		card = tempList.map((item, index) => {
			return <Card details={item} key={index} />;
		});
	}

	return (
		<>
			{/* NOTIFICATION BAR */}
			<div className="notification text-center py-4 md:px-9">
				Tada! Get started with a free template. Canâ€™t find what you are looking
				for? Search from the 1000+ available templates
			</div>

			{/* SUBNAV */}
			<div className=" mt-20 flex justify-between header ">
				<h5 className="text-md ">{tempCategory} Templates</h5>
				<h5 className="text-slate-400 text-sm">
					{finalData && finalData.length} Templates
				</h5>
			</div>

			{/* TEMPLATE LIST */}
			<div
				className=" grid md:grid-cols-3 lg:gap-x-24 md:gap-5 sm:grid-cols-2   cardList"
				data-testid={`test-1`}
			>
				{card}
			</div>

			{/* PAGINATION */}
			<Pagination curPage={curPage} pagNum={pagNum} pageChange={pageChange} />
		</>
	);
}

export default List;
