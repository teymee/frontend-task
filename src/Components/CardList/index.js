import React from "react";
import { useSelector } from "react-redux";

import "./card.css";
import NotFound from "./NotFound";
import List from "./List";

function CardList() {
	const { categorizedTemp } = useSelector(
		(state) => state.template
	);

	return (
		<>
			{categorizedTemp.length === 0 ? (
				<NotFound />
			):(
				<List />
			)}
		</>
	);
}

export default CardList;
