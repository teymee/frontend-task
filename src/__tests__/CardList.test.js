import { render, screen, cleanup } from "@testing-library/react";
import renderer from 'react-test-renderer'
import CardList from "../CardList/CardList";
import Card from '../CardList/Card'



import { Provider } from 'react-redux';
import store from "../store";

afterEach(()=>{
      cleanup()
})
test('Render CardList component', ()=>{
      render(
            <Provider store={store}>
      <CardList/>
      </Provider>)
      const cardListElement = screen.getByTestId('test-1')
      expect(cardListElement).toBeInTheDocument();
})

test('match snapshot', ()=>{
      const temp = {name:"laboris proident, velit", description:"proident, deserunt dolore cupidatat ullamco"}
      const tree = renderer.create(<Card name={temp.name} description={temp.description}/>).toJSON()
      expect(tree).toMatchSnapshot();
      
})