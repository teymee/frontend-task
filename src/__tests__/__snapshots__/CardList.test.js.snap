// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="card  my-10"
>
  <div
    className="container py-5"
  >
    <h1>
      laboris proident, velit
    </h1>
    <p
      className="my-5"
    >
      proident, deserunt dolore cupidatat ullamco
    </p>
  </div>
  <span
    className="pl-9 py-3"
  >
    Use Template
  </span>
</div>
`;
